packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -1433,33 +1426,29 @@ class CupertinoDialogRoute<T> extends RawDialogRoute<T> {
   CurvedAnimation? _fadeAnimation;
 
   @override
-  Widget buildTransitions(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget child) {
+  Simulation createSimulation({ required bool forward }) {
+    assert(!debugIsDisposed(), 'Cannot reuse a $runtimeType after disposing it.');
+    final double end = forward ? 1.0 : 0.0;
+    return SpringSimulation(_kStandardSpring, controller!.value, end, 0);
+  }
 
+  @override
+  Widget buildTransitions(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget child) {
     if (transitionBuilder != null) {
       return super.buildTransitions(context, animation, secondaryAnimation, child);
     }
 
-    if (_fadeAnimation?.parent != animation) {
-      _fadeAnimation?.dispose();
-      _fadeAnimation = CurvedAnimation(
-        parent: animation,
-        curve: Curves.easeInOut,
-      );
-    }
-
-    final CurvedAnimation fadeAnimation = _fadeAnimation!;
-
     if (animation.status == AnimationStatus.reverse) {
       return FadeTransition(
-        opacity: fadeAnimation,
-        child: super.buildTransitions(context, animation, secondaryAnimation, child),
+        opacity: animation,
+        child: child,
```

Comment:
why no super?

Reply:
The super's buildTransition is:
```dart
  Widget buildTransitions(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget child) {
    if (_transitionBuilder == null) {
      // Some default transition.
      return FadeTransition(
        opacity: animation,
        child: child,
      );
    }
    return _transitionBuilder(context, animation, secondaryAnimation, child);
  }
```
It's some default fading transition, and should not be applied on top of what we have here for cupertino route.

Reply:
(In fact I kind of think the 
```dart
    if (transitionBuilder != null) {
      return super.buildTransitions(context, animation, secondaryAnimation, child);
    }
```
part should not use super's buildTransitions either, but it's not the focus of this PR.)

Reply:
further looking at the code, is the intention that the CupertinoDialogRoute.transitionbuilder will override all transition in this class?

Reply:
> further looking at the code, is the intention that the CupertinoDialogRoute.transitionbuilder will override all transition in this class?

Yeah, that's how I comprehend the code before the PR. The case where `transitionBuilder` has been processed at the beginning of the method.

Reply:
sg